docs/Jest-integration.md

```diff
@@ -0,0 +1,100 @@
+# Jest integration
+
+In order to use Async Storage in your Jest tests, you need to [add implementation of its Native Module.](https://jestjs.io/docs/en/manual-mocks) 
+This library comes with a *simple*  mock (no actual functionality) that you can use. To add a custom behavior to it, checkout [customizing mock logic](#Custom-Mock-logic) section.
+
+
+## Using Async Storage mock
+
+You can use mocked Async Storage in one of two ways:
+
+### Mock `node_modules`
+
+1. In your project root directory, create `__mocks__` directory.
+2. Inside that folder, create `react-native.js` file.
+3. Inside that file, import `AsyncStorageMock` and use it to mock `NativeModules`.
+
+
+```js
+
+import AsyncStorageMock from '@react-native-community/async-storage/jest/async-storage-mock';
+
+jest.doMock('react-native', () => {
+  const reactNative = require.requireActual('react-native');
+
+  reactNative.NativeModules.RNCAsyncStorage = AsyncStorageMock;
+
+  return reactNative;
+});
+
+// to contain all other exports from RN
+export * from 'react-native';
+
+```
+
+### Use Jest setup files
+
+1. In Jest config (probably in `package.json`) add setup files location:
+
+```json
+// ...
+  "jest": {
+    // other config
+    "setupFiles": "./path/to/jestSetupFile.js"
+  },
+// ...
+
+```
+
+2. Inside your setup file, set up Async Storage mocking:
+
+
+```js
+
+import AsyncStorageMock from '@react-native-community/async-storage/jest/async-storage-mock';
+
+jest.doMock('react-native', () => {
+  const reactNative = require.requireActual('react-native');
+
+  reactNative.NativeModules.RNCAsyncStorage = AsyncStorageMock;
+
+  return reactNative;
+});
+
+```
+
+## Custom Mock logic
+
+`AsyncStorageMock` is a simple mock, without implemented functionality. If your test case relies on a storage, you can add you custom logic to it. [Go over a sample implementation in example app](../example/__mocks__/react-native.js) or see the code below:
+
+
+```js
+// somewhere in your configuration files
+import AsyncStorageMock from '@react-native-community/async-storage/jest/async-storage-mock';
+
+AsyncStorageMock.multiGet = jest.fn(([keys], callback) => {
+  // call callback with "retrieved" value after .multiGet call
+  callback();
+})
+
+export default AsyncStorageMock;
+
+```
+
+You can [check mock itself](../jest/async-storage-mock.js) to see each method's signatures.
+
+
+## Troubleshooting
+
+
+1. **`SyntaxError: Unexpected token export` in async-storage/lib/index.js**
+
+This is likely because `Jest` is not transforming Async Storage. You can point it to do so, by adding line in Jest's configuration:
+
+```json
+  "jest": {
+    // ...
+    "transformIgnorePatterns": ["/node_modules/@react-native-community/async-storage/(?!(lib))"]
```

Comment:
1. Make it valid JSON
2. Use a catch-all regex for all `@react-native-community` packages to help others :D :
```
"transformIgnorePatterns": ["node_modules/(?!react-native|@react-native-community)"]
```
3. Make a note that this should be included by default in RN 0.60

Reply:
I don't see this addressed, mind fixing?